import sys
from PyQt5 import QtWidgets
from PyQt5.QtWidgets import QDialog, QApplication, QMessageBox
from PyQt5.uic import loadUi
        
# ğŸ”´ Ø§Ù„ØªØ¹Ø¯ÙŠÙ„Ø§Øª Ø§Ù„Ø¬Ø¯ÙŠØ¯Ø© (Ø§Ù„Ø¥Ø¶Ø§ÙØ§Øª):
from Users import User                  # Ø§Ø³ØªÙŠØ±Ø§Ø¯ ÙƒÙ„Ø§Ø³ Ø§Ù„Ù…Ø³ØªØ®Ø¯Ù… Ù„Ù„ØªØ­Ù‚Ù‚ Ù…Ù† Ø§Ù„Ø¯Ø®ÙˆÙ„
from Students import Student            # Ø§Ø³ØªÙŠØ±Ø§Ø¯ ÙƒÙ„Ø§Ø³ Ø§Ù„Ø·Ø§Ù„Ø¨ Ù„Ø¥Ù†Ø´Ø§Ø¡ Ø·Ø§Ù„Ø¨ Ø¬Ø¯ÙŠØ¯
from student_dashboard import StudentDashboard  # Ø§Ø³ØªÙŠØ±Ø§Ø¯ Ø´Ø§Ø´Ø© Ø§Ù„Ø·Ø§Ù„Ø¨ Ù„Ù„Ø§Ù†ØªÙ‚Ø§Ù„ Ù„Ù‡Ø§
from admin_dashboard import AdminDashboard # Ø§Ø³ØªÙŠØ±Ø§Ø¯ Ø´Ø§Ø´Ø© Ø§Ù„Ø£Ø¯Ù…Ù† (Ù„Ùˆ Ø§Ø­ØªØ¬Ù†Ø§Ù‡Ø§ Ù„Ø§Ø­Ù‚Ø§Ù‹)

class Login(QDialog):
    def __init__(self,widget):
        super(Login,self).__init__()
        loadUi("ui/login.ui",self)
        self.widget=widget
        self.loginButton.clicked.connect(self.login)
        self.passwordInput.setEchoMode(QtWidgets.QLineEdit.Password)
        self.createButton.clicked.connect(self.open_signup)
    def login(self):
        email = self.emailInput.text()
        password = self.passwordInput.text()

        if email == "" or password == "":
            self.statusLabel.setText("Please enter email and password")
            return

        user = User.authenticate(email, password)

        if user:
            # -- Ø­Ø§Ù„Ø© Ø§Ù„Ù†Ø¬Ø§Ø­ --
            self.statusLabel.setText("Login Successful!")
            
            # Ù†ÙØ­Øµ Ø§Ù„Ø¯ÙˆØ±: Ù‡Ù„ Ù‡Ùˆ Ø·Ø§Ù„Ø¨ØŸ
            if user.role == "student":
                # Ù†ÙØªØ­ Ø´Ø§Ø´Ø© Ø§Ù„Ø·Ø§Ù„Ø¨ ÙˆÙ†Ù…Ø±Ø± Ù„Ù‡Ø§ Ø±Ù‚Ù… Ø§Ù„Ø·Ø§Ù„Ø¨ (Ù…Ù‡Ù… Ø¬Ø¯Ø§Ù‹ Ø¹Ø´Ø§Ù† ØªØ·Ù„Ø¹ Ø¨ÙŠØ§Ù†Ø§ØªÙ‡)
                self.dashboard = StudentDashboard(user.student_id) 
                self.dashboard.show()
                self.close() # Ù†Ù‚ÙÙ„ Ø´Ø§Ø´Ø© Ø§Ù„Ø¯Ø®ÙˆÙ„ Ø§Ù„Ø­Ø§Ù„ÙŠØ©
            else:
                self.statusLabel.setText("Welcome Admin (Dashboard not linked)")
        else:
            # -- Ø­Ø§Ù„Ø© Ø§Ù„ÙØ´Ù„ --
            self.statusLabel.setText("Invalid email or password")
    def open_signup(self):
        signup_window = signup(self.widget)
        self.widget.addWidget(signup_window)
        self.widget.setCurrentIndex(self.widget.currentIndex()+1)

class signup(QDialog):      
    def __init__(self,widget):
        self.widget=widget
        super(signup,self).__init__()
        loadUi("ui/signup.ui",self)
        self.signupButton.clicked.connect(self.signup)

    def signup(self):
        # 1. Ø³Ø­Ø¨ Ø§Ù„Ø¨ÙŠØ§Ù†Ø§Øª Ù…Ù† Ø§Ù„Ø­Ù‚ÙˆÙ„
        name = self.NameInput.text()
        student_id = self.IdInput.text()
        email = self.emailInput.text()
        program = self.programInput.text()
        level= self.levelInput.text()
        password = self.passwordInput.text()
        # ØªØ­ÙˆÙŠÙ„ Ø§Ù„Ù…Ø³ØªÙˆÙ‰ Ù„Ø±Ù‚Ù… (Ù…Ù‡Ù… Ø¹Ø´Ø§Ù† Ø§Ù„Ø¯Ø§ØªØ§ Ø¨ÙŠØ³)
        try:
            # ØªØ­ÙˆÙŠÙ„ Ø§Ù„Ù…Ø³ØªÙˆÙ‰ Ù„Ø±Ù‚Ù…
            level_int = int(level)

            # 1. Ø­ÙØ¸ Ø¨ÙŠØ§Ù†Ø§Øª Ø§Ù„Ø·Ø§Ù„Ø¨
            new_student = Student(student_id=student_id, name=name, email=email, program=program, level=level_int)
            new_student.save_to_db()

            # 2. Ø¥Ù†Ø´Ø§Ø¡ Ø­Ø³Ø§Ø¨ Ø¯Ø®ÙˆÙ„
            User.create_user(email=email, password=password, role="student", student_id=student_id)

            print("ØªÙ… Ø§Ù„ØªØ³Ø¬ÙŠÙ„ Ø¨Ù†Ø¬Ø§Ø­!")
            
            # Ø§Ù„Ø¹ÙˆØ¯Ø© Ù„Ø´Ø§Ø´Ø© Ø§Ù„Ø¯Ø®ÙˆÙ„
            #login_window = Login()
            #self.widget.addWidget(login_window)
            #self.widget.setCurrentIndex(self.widget.currentIndex()+1)
            self.widget.removeWidget(self)

        except Exception as e:
            print("Ø­Ø¯Ø« Ø®Ø·Ø£:", e)
#app = QtWidgets.QApplication(sys.argv)
#window = Login()
#widget=QtWidgets.QStackedWidget()
#widget.addWidget(window)
#widget.setFixedWidth(400)
#widget.setFixedHeight(500)
#widget.show()
#app.exec_()
